stl/inc/ranges

```diff
@@ -2462,6 +2462,333 @@ namespace ranges {
+        }
+
+        _NODISCARD constexpr _Iterator<false> begin() noexcept(
+            noexcept(_RANGES begin(_Range)) && is_nothrow_move_constructible_v<iterator_t<_Vw>>) /* strengthened */ {
+            return _Iterator<false>{*this, _RANGES begin(_Range)};
+        }
+
+        _NODISCARD constexpr _Iterator<true> begin() noexcept(
+            noexcept(_RANGES begin(_Range)) && is_nothrow_move_constructible_v<iterator_t<_Vw>>) /* strengthened */
+            requires _Simple_view<_Vw>&& is_reference_v<range_reference_t<_Vw>> {
+            return _Iterator<true>{*this, _RANGES begin(_Range)};
+        }
+
+        // clang-format off
+        _NODISCARD constexpr _Iterator<true> begin() const noexcept(noexcept(
+            _RANGES begin(_Range)) && is_nothrow_move_constructible_v<iterator_t<const _Vw>>) /* strengthened */
+            requires input_range<const _Vw> && is_reference_v<range_reference_t<const _Vw>> {
+            // clang-format on
+            return _Iterator<true>{*this, _RANGES begin(_Range)};
+        }
+
+        // clang-format off
+        _NODISCARD constexpr auto end() noexcept(noexcept(
+            _RANGES end(_Range)) && is_nothrow_move_constructible_v<sentinel_t<_Vw>>) /* strengthened */ {
+            if constexpr (forward_range<_Vw> && is_reference_v<range_reference_t<_Vw>>
+                      && forward_range<_InnerRng> && common_range<_Vw> && common_range<_InnerRng>) {
+                // clang-format on
+                return _Iterator<_Simple_view<_Vw>>{*this, _RANGES end(_Range)};
+            } else {
+                return _Sentinel<_Simple_view<_Vw>>{*this};
+            }
+        }
+
+        // clang-format off
+        _NODISCARD constexpr auto end() const noexcept(noexcept(
+            _RANGES end(_Range)) && is_nothrow_move_constructible_v<sentinel_t<_Vw>>) /* strengthened */
+            requires input_range<const _Vw> && is_reference_v<range_reference_t<const _Vw>> {
+            if constexpr (forward_range<const _Vw> && is_reference_v<range_reference_t<const _Vw>>
+                      && forward_range<range_reference_t<const _Vw>> && common_range<const _Vw>
+                      && common_range<range_reference_t<const _Vw>>) {
+                // clang-format on
+                return _Iterator<true>{*this, _RANGES end(_Range)};
+            } else {
+                return _Sentinel<true>{*this};
+            }
+        }
+    };
+
+    template <class _Rng>
+    join_view(_Rng &&) -> join_view<views::all_t<_Rng>>;
+
+    namespace views {
+        // VARIABLE views::join
+        class _Join_fn : public _Pipe::_Base<_Join_fn> {
+        public:
+            // clang-format off
+            template <viewable_range _Rng>
+            _NODISCARD constexpr auto operator()(_Rng&& _Range) const noexcept(noexcept(
+                join_view{_STD forward<_Rng>(_Range)})) requires requires {
+                join_view{static_cast<_Rng&&>(_Range)};

```

Comment:
Should this be:
```suggestion
                join_view<views::all_t<_Rng>>{_STD forward<_Rng>(_Range)})) requires requires {
                join_view<views::all_t<_Rng>>{static_cast<_Rng&&>(_Range)};
```

Reply:
Absolutely, thats what you get for having to write the same thing three times